samples/basic/blinky_pwm/boards/esp32.overlay

```diff
@@ -2,25 +2,42 @@
  * SPDX-License-Identifier: Apache-2.0
  *
  * Copyright (c) 2021 Andrei-Edward Popa
+ * Copyright (c) 2022 Espressif Systems (Shanghai) Co., Ltd.
  */
 
-#include <zephyr/dt-bindings/pwm/pwm.h>
+ #include <zephyr/dt-bindings/pwm/pwm.h>
+ #include <dt-bindings/pinctrl/esp32-pinctrl.h>
 
-/ {
-	aliases {
-		pwm-0 = &ledc0;
-		pwm-led0 = &pwm_led_blue;
-	};
+ / {
+     aliases {
+         pwm-0 = &ledc0;
+         pwm-led0 = &pwm_led_blue;
+     };
 
-	pwmleds {
-		compatible = "pwm-leds";
-		pwm_led_blue: pwm_led_gpio0_2 {
-		   label = "PWM LED0";
-		   pwms = <&ledc0 2 1000 PWM_POLARITY_NORMAL>;
-		};
-	};
+     pwmleds {
+         compatible = "pwm-leds";
+         pwm_led_blue: pwm_led_gpio0_2 {
+            label = "PWM LED0";
+            pwms = <&ledc0 0 1000 PWM_POLARITY_NORMAL>;
+         };
+     };
+ };
+
+&pinctrl {
+    ledc0_ch0_gpio2: ledc0_ch0_gpio2 {
+        pinmux = <LEDC_CH0_GPIO2>;
+        output-enable;
+    };
 };
```

Comment:
please use tabs, check your editor settings

Reply:
also, why isn't this defined in the board dts?

Reply:
1. Thanks for the tip. 
2. The esp32 is based on the esp32_devkitc board, wich has no built-in LED. Since the board itself has no native support for the LED I prefer not to define it in the dts, but to include this overlay so the regular user can run the sample with an external LED. I will include the board in the README of this sample with instructions to connect an external LED.